tests/foreman/ui/test_subscription.py

```diff
@@ -554,3 +556,82 @@ def test_positive_prepare_for_sca_only_subscription(target_sat, function_entitle
+        1. Upload expired manifest in newly created org
+        2. go to Content > Subscriptions page, click on 'Manage Manifest' button.
+        3. Search for message string 'Manifest expired', or 'Your manifest expired'
+        4. delete expired manifest from this org (cleanup part)
+        5. Upload non-expired manifest in default org
+        6. go to Content > Subscription page, click on 'Manage Manifest' button.
+        7. Search for message string 'Manifest expiring soon', or 'Your manifest will expire'
+        8. delete expired manifest from default org (cleanup part)
+
+    :expectedresults:
+        1. 'Manifest expired' message appear on screen
+
+    :BZ: 2075163
+
+    :customerscenario: true
+    """
+    property_name = setting_update.name
+    property_value = 366  # value should be greater than 365 to get expected output message
+    remote_path = f'/tmp/{EXPIRED_MANIFEST}'
+    target_sat.put(DataFile.EXPIRED_MANIFEST_FILE, remote_path)
+    # upload expired manifest
+    target_sat.cli.Subscription.upload({'organization-id': function_org.id, 'file': remote_path})
+
+    # Create new manifest
+    _, temporary_local_manifest_path = mkstemp(prefix='manifest-', suffix='.zip')
+    with clone() as manifest, open(temporary_local_manifest_path, 'wb') as file_handler:
+        file_handler.write(manifest.content.read())
+
+    with target_sat.ui_session() as session:
+        # Message - Manifest expired
+        session.organization.select(function_org.name)
+        expired_manifest = session.subscription.read_expire_manifest_message()
+        assert 'Manifest expired' in expired_manifest[0], 'Manifest expire alert not found'
+        assert ('Your manifest expired' and 'import a new manifest') in expired_manifest[1]
+        # Cleanup - delete expired manifest
+        session.subscription.delete_manifest(
+            ignore_error_messages=['Danger alert: Katello::Errors::UpstreamConsumerNotFound']
+        )
+
+        # Message - Manifest expiring soon
+        session.organization.select(default_org.name)
+        # Upload non-expire manifest
+        session.subscription.add_manifest(
+            temporary_local_manifest_path,
+            ignore_error_messages=['Danger alert: Katello::Errors::UpstreamConsumerNotFound'],
+        )
+        expiring_soon = session.subscription.read_expire_manifest_message()
+        if expiring_soon[0] is None:
+            # if 'Manifest expiring soon' message not found
+            # then update the 'Expire soon days' value from settings > Content
+            ### unable to search "name = 'expire_soon_days'" in web UI of Settings page ###
+            # session.settings.update(f'name = {property_name}', property_value)
+            ### in discussion with developer on the above issue, till then use CLI method ###
+            target_sat.cli.Settings.set({'name': f'{property_name}', 'value': f'{property_value}'})
+            session.browser.refresh()
+            time.sleep(5)
```

Comment:
Instead of adding sleep here, you can try adding `is_displayed` in read_expire_manifest_message to read `expired manifest message`

Reply:
yes, I was testing the changes locally, just update function flow again, please re-review again